lib/web_ui/flutter_js/src/loader.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import { detectBrowserEnvironment } from './browser_environment.js';
+import { FlutterEntrypointLoader } from './entrypoint_loader.js';
+import { FlutterServiceWorkerLoader } from './service_worker_loader.js';
+import { FlutterTrustedTypesPolicy } from './trusted_types.js';
+import { loadCanvasKit } from './canvaskit_loader.js';
+import { loadSkwasm } from './skwasm_loader.js';
+
+/**
+ * The public interface of _flutter.loader. Exposes two methods:
+ * * loadEntrypoint (which coordinates the default Flutter web loading procedure)
+ * * didCreateEngineInitializer (which is called by Flutter to notify that its
+ *                              Engine is ready to be initialized)
+ */
+export class FlutterLoader {
+  /**
+   * @deprecated Use `load` instead.
+   * Initializes the Flutter web app.
+   * @param {*} options
+   * @returns {Promise?} a (Deprecated) Promise that will eventually resolve
+   *                     with an EngineInitializer, or will be rejected with
+   *                     any error caused by the loader. Or Null, if the user
+   *                     supplies an `onEntrypointLoaded` Function as an option.
+   */
+  async loadEntrypoint(options) {
```

Comment:
Is `load` a superset of `loadEntrypoint`? Can `loadEntrypoint` be re-implemented as a config+call to `load`?

Reply:
I contemplated (and made an attempt at) doing this, but it's complicated because they are definitely subtly different in ways that are hard to reconcile cleanly. For example, `loadEntrypoint` only works if it knows about the build configuration, but `loadEntrypoint` actually doesn't know anything about certain parts of the build configuration (i.e. what renderer this build is using). So `loadEntrypoint` would have to make up a bogus build configuration (like saying it's the `html` renderer in order to prevent `load` from attempting to pre-initialize CanvasKit and so on) and it just feels weird and wrong. In general, I thought maybe just to keep things stable, I would try to keep this code path mostly only mechanically refactored, rather than rewritten. Hopefully we can remove it before too long anyway.

Reply:
ACK, thanks for considering it though!